File: src/main/java/net/starfal/kvisuals/Functions/Color.java
package net.starfal.kvisuals.Functions;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
import org.bukkit.ChatColor;

public class Color {
    public static String format(String miniMessage) {
        if (miniMessage == null) {
            return null;
        }
        MiniMessage mm = MiniMessage.miniMessage();
        Component message = mm.deserialize(miniMessage);
        String legacyMessage = LegacyComponentSerializer.legacyAmpersand().serialize(message);
        return ChatColor.translateAlternateColorCodes('&', legacyMessage);
    }
}

File: src/main/kotlin/net/starfal/funs/Color.kt
package net.starfal.funs

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage

class Color {
    companion object {
        fun format (text: String): Component {
            return MiniMessage.miniMessage().deserialize(text)
        }
    }
}